src/Source/PropelCollectionSourceIterator.php

```diff
@@ -39,13 +39,10 @@ public function __construct(PropelCollection $collection, array $fields, string
 
     public function rewind(): void
     {
-        if ($this->iterator) {
-            $this->iterator->rewind();
-
-            return;
+        if (!$this->iterator) {
+            $this->iterator = $this->collection->getIterator();
         }
 
-        $this->iterator = $this->collection->getIterator();
```

Comment:
No exception was thrown previously here… would it be fair to say that you could already call `rewind()` multiple times here? If yes, how does what are you doing here exactly, and how does it relate to this commit?